src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ConverterList.cs

```diff
@@ -20,6 +20,12 @@ public ConverterList(JsonSerializerOptions options)
             _options = options;
         }
 
+        public ConverterList(JsonSerializerOptions options, ConverterList source)
+        {
+            _options = options;
```

Comment:
If we manually copy the elements instead, then we can set the capacity of the dest List to the source's `.Count`. This will avoid extra allocations in the dest List's internal array.